src/Engine/Engine.cpp

```diff
@@ -1458,8 +1458,9 @@ void _494035_timed_effects__water_walking_damage__etc() {
     pParty->uCurrentMonth = pParty->GetPlayingTime().GetMonthsOfYear();
     pParty->uCurrentYear = pParty->GetPlayingTime().GetYears() + game_starting_year;
 
-    // TODO(pskelton): Condition looks wierd, investigate, possible cause of #504
-    if (pParty->uCurrentHour >= 3 && (old_hour < 3 || pParty->uCurrentDayOfMonth > old_day)) {  // new day dawns
+    // New day dawns
+    // TODO(pskelton): ticks over at 3 in the morning?? check
+    if ((pParty->uCurrentYear > old_year) || pParty->uCurrentHour >= 3 && (old_hour < 3 || pParty->uCurrentDayOfMonth > old_day)) {
```

Comment:
I think we need smth like this here:
```
int oldHours = pParty->GetPlayingTime().GetHours();

pParty->GetPlayingTime().value += pEventTimer->uTimeElapsed;

int newHours = pParty->GetPlayingTime().GetHours();
int lastNight = (newHours - 3) / 24 * 24 + 3; // Last time the clock showed 3am

if (oldHours < lastNight && newHours >= lastNight) { // We've jumped over the last 3am boundary
```

Reply:
The function is not called in a way that guaranties it would straddle the tickover. I need to think some more.

Reply:
A neater solution would be to store GetDays() in an unused party var. Party time can be advanced elsewhere than this func so weve got to really on a time that gets saved somewhere. Will add a TODO.

```
int newdays = pParty->GetPlayingTime().GetDays();

if (newdays > pParty->_currentDays && pParty->uCurrentHour >= 3) {
        pParty->_currentDays = newdays;
```